### PR TITLE
Handle empty label selectors in web console

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@liggitt

File: assets/app/scripts/misc/labelSelector.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
-function LabelSelector(selector) {
+// selector (optional) - the JSON format as returned by k8s API
+// emptySelects (optional) - whether a label selector with no conjuncts
+//      selects objects.  Typical behavior is false.  Example of an
+//      exceptional case is when filtering by labels, no label selectors
+//      means no filters.
+function LabelSelector(selector, emptySelects) {
```
Comment (liggitt) [inline assets/app/scripts/misc/labelSelector.js]: name `emptySelectsAll`? `emptySelects` sounds like it is a list of the things `empty` selected

File: assets/app/scripts/misc/labelSelector.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
   this._conjuncts = {};
+  this._emptySelects = emptySelects;
```
Comment (liggitt) [inline assets/app/scripts/misc/labelSelector.js]: cast to boolean with `!!`?

File: assets/app/scripts/misc/labelSelector.js [hunk 3]
```diff
@@ -61,6 +67,10 @@ LabelSelector.prototype.matches = function(resource) {
   if (!resource) {
     return false;
   }
+  // "null" label selectors don't match anything
+  if (!this._emptySelects && this.isEmpty()) {
+    return false;
+  }
```
Comment (liggitt) [inline assets/app/scripts/misc/labelSelector.js]: need to return true or false if isEmpty()
```
if (this.isEmpty()) {
  return this._emptySelectsAll;
} 
```

File: assets/app/scripts/misc/labelSelector.js [hunk 4]
```diff
@@ -107,6 +117,11 @@ LabelSelector.prototype.hasConjunct = function(conjunct) {
 
 // Test whether this label selector covers the given selector
 LabelSelector.prototype.covers = function(selector) {
+  // "null" label selectors don't match anything
+  if (!this._emptySelects && this.isEmpty()) {
+    return false;
+  }
```
Comment (liggitt) [inline assets/app/scripts/misc/labelSelector.js]: same comment as above about returning true or false. does an empty label selector cover another empty label selector? what if their `_emptySelectsAll` flag is different?

File: assets/app/scripts/misc/labelSelector.js [hunk 5]
```diff
@@ -1,5 +1,11 @@
-function LabelSelector(selector) {
+// selector (optional) - the JSON format as returned by k8s API
+// emptySelects (optional) - whether a label selector with no conjuncts
```
Comment (liggitt) [inline assets/app/scripts/misc/labelSelector.js]: param name

**Comment by liggitt** [comment]
doc nit, then LGTM

**Comment by jwforres** [comment]
[merge]

**Pull request merged**